src/main/java/uk/co/ramp/covid/simulation/RStats.java

```diff
@@ -31,21 +32,21 @@ private Double getMeanRGeneric(int absDay, BiFunction<Integer, Integer, Boolean>
             }
         }
 
-        if (nSecondaries > 0) {
-            return nSecondaries /  nInfectors;
-        }
-
+        if(nInfectors > 0 && !rflag) return nSecondaries;
+        if(rflag) return nSecondaries / nInfectors;
```

Comment:
I think the nInfectors > 0 check should be on the next line with "rflag" to avoid the division by 0

Reply:
The first point, I have left this as one method that spits outone of two answers depending on a flag because I think that his will be revisited, so I don't want to be rewriting two different methods